src/main/java/uk/gov/hmcts/reform/divorce/orchestration/tasks/RemoveListingDataTask.java

```
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.divorce.orchestration.tasks;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.divorce.orchestration.domain.model.ccd.CollectionMember;
+import uk.gov.hmcts.reform.divorce.orchestration.framework.workflow.task.Task;
+import uk.gov.hmcts.reform.divorce.orchestration.framework.workflow.task.TaskContext;
+import uk.gov.hmcts.reform.divorce.orchestration.framework.workflow.task.TaskException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.BulkCaseConstants.COURT_NAME_CCD_FIELD;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.BULK_LISTING_CASE_ID_FIELD;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.DATETIME_OF_HEARING_CCD_FIELD;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.PRONOUNCEMENT_JUDGE_CCD_FIELD;
+
+@Component
+public class RemoveListingDataTask implements Task<Map<String, Object>> {
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Override
+    public Map<String, Object> execute(TaskContext context, final Map<String, Object> payload) throws TaskException {
+        Map<String, Object> response = new HashMap<>(payload);
+        response.remove(BULK_LISTING_CASE_ID_FIELD);
+        response.remove(COURT_NAME_CCD_FIELD);
+        response.remove(PRONOUNCEMENT_JUDGE_CCD_FIELD);
+
+        List<CollectionMember<Map<String, Object>>> listOfCourtHearings =
+                mapper.convertValue(payload.get(DATETIME_OF_HEARING_CCD_FIELD), ArrayList.class);
+        listOfCourtHearings.remove(listOfCourtHearings.size() - 1);
```

I hope it's not possible to get index out of bound here?

---

If we don't mess up, it shouldn't be possible. The case can only be in the state which triggers this event after it has been listed for court hearing, so there must be at least 1 element.

Is it safer to catch a OutOfBounds / NPE or just let it throw an error at runtime?